rust/datafusion/src/physical_plan/repartition.rs

```diff
@@ -138,8 +138,8 @@ impl ExecutionPlan for RepartitionExec {
                     }
 
                     // notify each output partition that this input partition has no more data
-                    for i in 0..num_output_partitions {
-                        let tx = &mut channels[i].0;
+                    for channel in channels.iter_mut().take(num_output_partitions) {
```

Comment:
This can probably be even simpler as I think `channels` always has `num_output_partitions` elemements: 
```suggestion
                    for channel in channels.iter_mut() {
```

But this PR's code is a correct as what was previously here :)